engine_modules/corporation/models.py

```diff
@@ -92,20 +96,39 @@ class Meta:
 
 	base_corporation_slug = models.CharField(max_length=20)
 	game = models.ForeignKey(Game)
-	assets = models.SmallIntegerField()
-	historic_market = models.ForeignKey(Market)
+	assets = models.SmallIntegerField(default=0)
+	market_assets = models.SmallIntegerField()
+	assets_modifier = models.SmallIntegerField(default=0)
+
+	@property
+	def corporation_markets(self):
+		return CorporationMarket.objects.filter(corporation=self)
+
+	@property
+	def markets(self):
+		return [cm.market for cm in self.corporation_markets]
+
+	def get_common_market(self, c2):
+		# Returns the CorporationMaket for a common market if there is one, None if not
+                common_market = None
+		common_markets = []
+
+                for cm in self.corporation_markets:
+                        if cm.market in [cm2.market for cm2 in c2.corporation_markets]:
+                                common_markets.append(cm)
+		if len(common_markets) != 0:
```

Comment:
I do not agree, I think returning a random market has much more chances of uncovering potential bugs.
Applications that need the common market to be a specific one must enforce it themselves, instead of relying on an unspoken mechanism.